### ISSUE TITLE
Room Owner Role is Removed From Owner

### ISSUE THREAD

**Issue by stickboy06** [RocketChat/Rocket.Chat#3263]
I'm running server version 0.30.0, but I've seen this issue with multiple versions.

The problem that I am seeing is that a user will create a private group and become the owner.  At some point, the role of owner gets removed from that user.  This then leaves no one in the room that can add users or change the roles of current users.  This requires an admin to go into the Administration page, set the private group to a public one and then go in and change the user roles so they can administer the room again, and then set the room back to private.  This has happened three or four times and I believe it was always with private groups.  When this happens, I have the owner set another user as moderator so they can fix this issue if it happens again.  It may be happening more than I know about because of this.

**Comment by marceloschmidt** [comment]
Can you reproduce this with ease? Can you try creating a private group on demo.rocket.chat and check if tat will happen there as well? Thanks!

**Comment by stickboy06** [comment]
I found another room today with no owner on our instance of RocketChat.  I am unable to reproduce it on demo.rocket.chat.  I've also noticed that the rooms where this has happened are older rooms created months ago.  I don't think I've seen it on newer rooms, but I can't be sure.  Is there a way to find out when a room was created?  Maybe there was an earlier bug where rooms were created with no owners.

**Comment by marceloschmidt** [comment]
Can you check if the rooms without any owner have at least a moderator? I've just checked our migrations and on migration v19 we added moderators to all rooms. I realize that it's possible that the moderator might have left the room already... if it's too much of a trouble, we could create a script that would set room creators as owners of rooms for you.

**Comment by stickboy06** [comment]
So far, the times I have been informed about the issue was when the room had no moderator.  It could have happened in rooms with moderators but the users just fixed it themselves, or it might not.  I can't say for sure.  It seems that in order to check for owners of private rooms, I have to go into the room and check the users for owners.  This requires me to make the room public first.  Is there an easier way to view room owners?  It would be nice for administrators to be able to change owner/moderator from the rooms tab in the administration menu.

**Comment by FibreFoX** [comment]
Happened to me some seconds ago ... i created some channel (because of evaluation of rocket.chat), some members entered that channel and used it productively.

After some users used it productively, I left the channel, not recognizing the message about "make sure to promote someone as mod"-message. After finding this out, I went into the administration-interface, headed to the "roles"-window, clicked on "owner", searched for my channel to modify ... then I tried to select another user ... clicked on the "add"-button and got some confusing message (sorry, it is german).
![rocket chat-issue-3263](https://cloud.githubusercontent.com/assets/1042579/17805210/9e4b2cd4-65fe-11e6-95a5-41bef232d8e2.PNG)

Using Rocket.Chat server 0.37.0, with rocket.chat windows-client 1.3.1

The really confusing part: after that message, I wasn't allowed to join the channel "/join #myChannelName" said "the channel does not exist", even I was able to "see" the channel.

After waiting for some time, another user send a message to that chat room ... and now I am again the owner, and joined to that room.

Is it possible that some async request internally isn't fast enough? Or does it recover from this?

(Sorry if this is not the correct issue ... but I think it is related)
